app/javascript/components/ui/Sheet.tsx

```diff
@@ -0,0 +1,98 @@
+import cx from "classnames";
+import * as React from "react";
+
+import { Icon } from "$app/components/Icons";
+
+type SheetProps = {
+  children: React.ReactNode;
+  className?: string;
+  ariaLabel?: string;
+  onClose?: () => void;
+  header?: React.ReactNode;
+};
+
+type SheetHeaderProps = {
+  children: React.ReactNode;
+  className?: string;
+  onClose?: () => void;
+};
+
+type SheetTitleProps = {
+  children: React.ReactNode;
+  className?: string;
+};
+
+type SheetFooterProps = {
+  children: React.ReactNode;
+  className?: string;
+};
+
+type SheetCloseProps = {
+  children: React.ReactNode;
+  asChild?: boolean;
+  onClick?: React.MouseEventHandler<HTMLElement>;
+};
+
+export const Sheet = ({ children, className, ariaLabel, onClose, header }: SheetProps) => {
```

Comment:
Think it would make sense to use `@radix-ui/react-dialog` here too the way shadcn does, see `Modal.tsx` for an example